### PR TITLE
Milestone 3

### PR DESCRIPTION
**Pull request by bruna-genz**
In this Milestone, we set up the authentication with the Devise gem. Also, we created User and Post models with validations and associations. Finally, we also create posts, dashboards and contents controller and the required actions to make our app functional.
We use TDD approach, with unit and integration tests.

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -9,7 +9,23 @@
     <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
```
Comment (maelfosso) [inline app/views/layouts/application.html.erb]: I can't manage to start the project because of that line.

**Comment by wathigo** [comment]
### Invalid PR
The line mentioned by @maelfosso  is still throwing an error while starting the app.
We don't have `app/assets/javascripts/`  folder.
Please fix this and submit another review request.
Screenshot from 2020-01-20 13-50-01

**Comment by bruna-genz** [comment]
We made some changes and now is everything working well on our computers.

**Comment by bruna-genz** [comment]
About Mael requests:

1. We have a link to logout in the navbar that is working. 
2. We now have a list of all the users on the link Fiend Friends in the navbar.
3. We now fixed the problem with /posts. It redirects to the sign in page.

**Comment by bruna-genz** [comment]
Sorry Ebuka, we were in the middle of making these changes. Now everything is there.

Thanks!

**Comment by bruna-genz** [comment]
Thanks @Redvanisationfot the review, we're gonna make all these changes.

We just couldn't understand why is necessary to have the `@posts = Post.all` in the `PostsController#index`. We did exactly this, but in the `DashboardsController#show` and it's working as a timeline. 
We thought that creating a new controller to manage these pages that handles a lot of other controllers (users, posts, comments, likes) would be easier, is that wrong to do?

**Comment by Redvanisation** [comment]
Hey @bruna-genz the rails convention is the use CRUD actions in controllers, and to list all the records is a task to be covered done in the `index` action of the post, [this article](https://stackify.com/what-are-crud-operations/) explains why CRUD is important.

If I had to refactor the project I would separate the controllers make each one responsible only for its own table in the DB then I'd make some helper methods and use them to display different things on the views.

_**That suggestion to refactor is only my personal opinion**._

**Comment by nicolasmarino99** [comment]
Thank you so much @Redvanisation for your helpful explanation. We have refactor all the required suggestions.

File: app/assets/stylesheets/contents/new.scss [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    position: absolute;
+    bottom: 0;
+    display: flex;
+    flex-direction: row;
+    list-style: none;
+    background-color: white;
+    text-decoration: none;
+    font-size: 8px;
+
+}
```
Comment (uche-inyama) [inline app/assets/stylesheets/contents/new.scss]: remember to remove commented code.

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -1,2 +1,20 @@
...
+        devise_parameter_sanitizer.permit(:account_update, keys: [:first_name, :last_name, :birthday, :gender])
+    end
+
+    private
+    
+        #def authenticate_user!
+        #    unless user_signed_in?
+        #        redirect_to new_content_path
+        #    end
+        #end
```
Comment (uche-inyama) [inline app/controllers/application_controller.rb]: Remember to delete commented code.

**Pull request merged**